### PR TITLE
New Dependency Fetch Parameter. Solves #4 [semver:minor]

### PR DESCRIPTION
**Pull request by KyleTryon**
# New parameter

`fetch_depenencies_command`
```
fetch_depenencies_command:
    description: Specify the command used to fetch dependencies.
    type: string
    default: "dependency:go-offline"
```

## Why?

Issue: #4 
@radek-svanda-elder has done an excellent job detailing the issue here. `dependency:go-offline` may not be the ideal command for fetching dependencies for all users. This new parameter will not affect current users but will allow the dependency fetch command to be overwritten.

# Also:

Added git tag releases to CI

### CODE REVIEW

File: src/commands/with_cache.yml [hunk 1]
```diff
@@ -26,6 +26,10 @@ parameters:
     description: Useful when the source of your maven project is nott in the root directory of your git repo. Supply the name of the directory or relative path of the directory containing your source code.
     type: string
     default: ''
+  fetch_depenencies_command:
```
Comment (lokst) [inline src/commands/with_cache.yml:29]: Should this be `fetch_dependencies_command` ?

File: src/jobs/parallel_test.yml [hunk 2]
```diff
@@ -23,6 +23,10 @@ parameters:
     description: Specify a custom path for invoking maven
     type: string
     default: mvn
+  fetch_depenencies_command:
```
Comment (joekewl70) [inline src/jobs/parallel_test.yml:26]:
```suggestion
  fetch_depenencies_command:.gitignore
```

**Comment by joekewl70** [comment]
There's still no way of helping me with a BTC mining app or program to save my 6 yr  daughter from becoming a foster child eh? It's to the point the ministry is going to place her with a junkie thats abusive to her,  a person that hates me because I won't lie for her, or a Foster family....

They've admitted they caused me to be homeless, jobless, and made me severely depressed by taking her away from her mother and not allowing me access in the beginning....yet they refuse to correct their mistake.....there us absolutely no assistance for single fathers.....no resources at all....I can't get back up to a level suitable to raise a beautiful 6 year old daughter.....I need drastic help, I can't allow her to be raised by foster parents.....

If you can find it in your heart to help me save my daughter my BTC address is


bc1q5yyv2m5sz4d3sdjg7yh6238vx60y6p0ra2qscf


Thank you for your time


Joseph Hebert

**Comment by gmemstr** [comment]
There are some worthwhile changes in this, but they are out of scope of the core issue. Closing this but hopefully we can revive some of the changes.

**Pull request closed without merging**